### PR TITLE
Slice of structs support

### PR DESCRIPTION
**Pull request by SpectrumQT**
As for me, auto-iterating over slices of structs feels more like expected behaviour. Please review.

Original issue: #9

### CODE REVIEW

File: defaults.go [hunk 1]
```diff
@@ -157,6 +157,20 @@ func setField(field reflect.Value, defaultVal string) error {
 		}
 		callSetter(ref.Interface())
 		field.Set(ref.Elem())
+	case reflect.Slice:
+		for j := 0; j < field.Len(); j++ {
+			sliceItem := field.Index(j)
+			if sliceItem.Kind() != reflect.Struct {
```
Comment (creasty) [inline defaults.go]: Might want to call `setField` recursively?

Comment (SpectrumQT) [reply defaults.go]: Yes, definitely. Updated commit.

**Comment by creasty** [comment]
Oh, let me fix the CI. It seems to be outdated...

```
$ go get -u github.com/golang/lint/golint
# golang.org/x/tools/go/internal/gcimporter
../../../golang.org/x/tools/go/internal/gcimporter/bexport.go:212: obj.IsAlias undefined (type *types.TypeName has no field or method IsAlias)
The command "go get -u github.com/golang/lint/golint" failed and exited with 2 during .
```

https://travis-ci.org/creasty/defaults/builds/485234065?utm_source=github_status&utm_medium=notification

**Comment by creasty** [comment]
Meanwhile, could you add test cases?

**Comment by SpectrumQT** [comment]
> Please make sure it passes CI.
If I get it right, defaults_test.go tests fails and I should edit it? Sorry, I'm new to this stuff.

**Comment by creasty** [comment]
> If I get it right, defaults_test.go tests fails and I should edit it?

Not sure if it will break the existing test cases. But, either way, you'll need to edit it.

**Comment by creasty** [comment]
> Oh, let me fix the CI. It seems to be outdated...

Fixed here https://github.com/creasty/defaults/pull/11

**Comment by SpectrumQT** [comment]
Fine, I've implemented new test for this feature and ensure passage of the old ones. Please take a look.

**Comment by creasty** [comment]
Released: https://github.com/creasty/defaults/releases/tag/v1.3.0

**Comment by SpectrumQT** [comment]
Was glad to contribute. Thanks for your little handy lib, my utility's config really loved it!

**Pull request merged**